File: articles/al-kindi-frequency-analysis-and-the-birth-of-cryptanalysis/go/frequency_count.go
package main

import (
	"fmt"
	"strings"
	"unicode"
)

func main() {
	text := "THIS IS A TOY CLASSICAL CIPHER EXAMPLE"
	counts := map[rune]int{}
	for _, ch := range strings.ToLower(text) {
		if unicode.IsLetter(ch) {
			counts[ch]++
		}
	}
	fmt.Println(counts)
}

File: articles/al-kindi-frequency-analysis-and-the-birth-of-cryptanalysis/cpp/frequency_count.cpp
#include <algorithm>
#include <cctype>
#include <iostream>
#include <map>
#include <string>

int main() {
    std::string text = "THIS IS A TOY CLASSICAL CIPHER EXAMPLE";
    std::map<char, int> counts;
    for (char ch : text) {
        unsigned char uch = static_cast<unsigned char>(ch);
        if (std::isalpha(uch)) {
            counts[static_cast<char>(std::tolower(uch))]++;
        }
    }
    for (const auto& [symbol, count] : counts) {
        std::cout << symbol << "," << count << "\n";
    }
    return 0;
}
